scale/scheduler/scale_scheduler.py

```diff
@@ -256,6 +256,8 @@ def statusUpdate(self, driver, status):
                     running_job_exe.task_running(task_id, results.when)
                 elif status.state == mesos_pb2.TASK_FINISHED:
                     running_job_exe.task_complete(results)
+                    if results.exit_code is None or not type(results.exit_code) is type(int):
+                        results.exit_code = 0
```

Comment:
I wasn't sure about the type guarantee for the exit_code. I dug into the mesos code a bit more and it looks like it's going to be an int if it's there so I think we're safe removing the check.